app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CartProductsController < ApplicationController
+  before_action :auth_check
+  before_action :load_cart, only: %i( new create edit update destroy)
+
+  def new
+    @cart_product = @cart.cart_products.new
+  end
+
+  def create
+    @cart_product = @cart.cart_products.new(cart_params)
+    @cart_product.cart_product_total = @cart_product.product.price * @cart_product.quantity
```

Comment:
Don't use table name prefix for attributes. Its should be just `@cart_product.total`